diff --git a/FsCrypt.cpp b/FsCrypt.cpp
index 7505e56..294fa27 100644
--- a/FsCrypt.cpp
+++ b/FsCrypt.cpp
@@ -515,8 +515,7 @@ bool fscrypt_init_user0() {
     // entangled with user credentials.  The framework will always prepare CE
     // storage once CE keys are installed.
     if (!fscrypt_prepare_user_storage("", 0, 0, android::os::IVold::STORAGE_FLAG_DE)) {
-        LOG(ERROR) << "Failed to prepare user 0 storage";
-        return false;
+        LOG(ERROR) << "Failed to prepare user 0 storage (non-fatal)";
     }
 
     // If this is a non-FBE device that recently left an emulated mode,
